ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/SurrogateTemplateIdCache.scala

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http.dbbackend
+
+import com.codahale.metrics.MetricRegistry
+import com.daml.caching.SizedCache
+import com.daml.http.dbbackend.Queries.SurrogateTpId
+import com.daml.metrics.Metrics
+import org.slf4j.LoggerFactory
+
+object SurrogateTemplateIdCache {
+  private val logger = LoggerFactory.getLogger(classOf[SurrogateTemplateIdCache])
+  private val maxSize = 1000L
+}
+
+trait SurrogateTemplateIdCache {
+  import SurrogateTemplateIdCache.{logger, maxSize}
+
+  private val metrics = new Metrics(new MetricRegistry)
```

Comment:
I believe the JSON API already has metrics, we may want to integrate into that.

Reply:
As a general suggestion, I would recommend against having `val`s in `trait`s to prevent initialization order issues.